mslib/mswms/gallery_builder.py

```diff
@@ -0,0 +1,363 @@
+    return f"""<div class="gallery">
+                 {image}
+                 <div class="gtooltip">
+                  {caption}<span class="gtooltiptext">{tooltip}</span>
+                 </div>
+                </div>"""
+
+
+def write_doc_index():
+    """
+    Write index containing all code examples for the sphinx docs
+    """
+    with open(os.path.join(docs_location, "code", "index.rst"), "w+") as rst:
+        files = "\n".join(sorted(["   " + f[:-4] for f in os.listdir(os.path.join(docs_location, "code"))
+                                  if "index" not in f and ".rst" in f]))
+        rst.write(f"""
+Code Examples
+--------------
+
+.. toctree::
+
+{files}
+""")
+
+
+def write_js(sphinx=False):
+    """
+    Writes the plots.js file containing the gallery
+    """
+    location = docs_location if sphinx else os.path.join(static_location, "docs")
+    js = begin
+    if sphinx:
+        js = js.replace("<h3>Plot Gallery</h3>", "")
+
+    for l_type in plots:
+        style = ""
+        if l_type == "Top":
+            style = "style=\"display: block;\""
+        js += f"<div id=\"{l_type} View\" class=\"tabcontent\" {style}>"
+        js += "\n".join(plots[l_type])
+        js += "</div>"
+
+    with open(os.path.join(location, "plots.js"), "w+") as md:
+        md.write(js + end)
+
+
+def write_plot_details(plot_object, l_type="top", sphinx=False):
+    """
+    Extracts and writes the plots code files at static/code/*
+    """
+    layer = "horizontal" if l_type == "Top" else "vertical" if l_type == "Side" else "linear"
+    location = docs_location if sphinx else static_location
+
+    if not os.path.exists(os.path.join(location, "code")):
+        os.mkdir(os.path.join(location, "code"))
+
+    modules = [m for m in inspect.getsource(inspect.getmodule(type(plot_object))).splitlines()
+               if m.startswith("import") or m.startswith("from")]
+
+    if sphinx:
+        write_plot_details_sphinx(plot_object, l_type, layer, modules)
+        return
+
+    with open(os.path.join(location, "code", f"{l_type}_{plot_object.name}.md"), "w+") as md:
+        md.write(f"**How to use this plot**  \n1. Make sure you have the required datafields "
+                 f"({', '.join(f'`{field[1]}`' for field in plot_object.required_datafields)})  \n"
+                 f"2. [Download this file](/mss/code/{l_type}_{plot_object.name}.md?download=True)  \n"
+                 f"3. Put this file into your mss_wms_settings.py directory, e.g. `~/mss`  \n"
+                 f"4. Append this code into your `mss_wms_settings.py`:  \n")
+        md.write(f"---\n```python\nfrom {l_type}_{plot_object.name} import {plot_object.__class__.__name__}\n"
+                 f"register_{layer}_layers = [] if not register_{layer}_layers else register_{layer}_layers\n"
+                 f"register_{layer}_layers.append(({plot_object.__class__.__name__}, [next(iter(data))]))\n```\n---\n")
```

Comment:
"next(iter(data)) " does this work for any server?

Reply:
It takes the first key of the data defined in the mss_wms_settings.py, so for most cases it works fine.
If there are more than one datasets and the plot doesn't derive from the first one, the user should change it.
Alternatively one could just list all datasets but I feel like that would be a wasteful solution.